assets/src/edit-story/components/header/buttons.js

```diff
@@ -55,19 +59,94 @@ function PreviewButton() {
       meta: { isSaving },
       story: { link },
     },
+    actions: { saveStory },
   } = useStory();
 
+  const [previewLinkToOpenViaDialog, setPreviewLinkToOpenViaDialog] = useState(
+    null
+  );
+
   /**
    * Open a preview of the story in current window.
    */
   const openPreviewLink = () => {
     const previewLink = addQueryArgs(link, { preview: 'true' });
-    window.open(previewLink, 'story-preview');
+
+    // Start a about:blank popup with waiting message until we complete
+    // the saving operation. That way we will not bust the popup timeout.
+    let popup;
+    try {
+      popup = window.open('about:blank', PREVIEW_TARGET);
+      if (popup) {
+        popup.document.write('<!DOCTYPE html><html><body>');
+        // Output "waiting" message.
+        popup.document.write(
+          escapeHTML(
+            __('Please wait. Generating the preview...', 'web-stories')
+          )
+        );
+        // Force redirect to the preview URL after 5 seconds. The saving tab
+        // might get frozen by the browser.
+        popup.document.write(
+          `<script>
+            setTimeout(function() {
+              location.replace(${JSON.stringify(previewLink)});
+            }, 5000);
+          </script>`
+        );
+      }
+    } catch (e) {
+      // Ignore errors. Anything can happen with a popup. The errors
+      // will be resolved after the story is saved.
+    }
+
+    // DO NOT SUBMIT: what if the status is no longer draft?
```

Comment:
Can't we just keep the current status as-is?

Reply:
Yep, previewing should definitely not modify the status.

Reply:
I definitely agree with this in theory. But it bugs me that in effect the button that says "Preview"  will in effect re-publish the new story changes.

Reply:
I can understand your concern, as it means unwittingly altering an already published story. But converting to draft is even worse - it means that the published story is now completely removed from public view instead.

@swissspidy, what does GB do here? I assume you can preview a published post without altering the already published post? Is it using revisions, creating a new draft revision to preview and keeping the old revision as the published one?

Reply:
Here's how I'd see this ideally, but no idea if WP supports this at all:
 1. There are two versions of a document: draft and last published. 
 2. After last publish: draft == published.
 3. If intermittent changes are needed, the draft is updated. But it doesn't change what's published is. I.e. draft > published.
 4. draft > publish is a clear signal that the document has unpublished changes since last publish.
 5. User can easily push the last draft to publish. In this case we're back to draft = publish.

/to @pbakaus, assuming the about is not possible to too hard, are you ok for "preview" to auto-publish changes?

Reply:
The above (the flow Dima documented) seems like the right course of action. I also wonder what Gutenberg is doing here - they can't possibly re-publish on preview?

Reply:
Moved this discussion to https://github.com/google/web-stories-wp/issues/1149

Reply:
I changed to preserve the existing status and tagged the issue. For now it's a lesser evil.